### ISSUE TITLE
nh_total_btc and nh_btc_usd_rate are always 0

### ISSUE THREAD

**Issue by esille** [tg44/nicehash-exporter#1]
Hi,

I'm using docker image and noticed that nh_total_btc and nh_btc_usd_rate are always 0. After checking logs in container i see following error reported:

```
(node:1) UnhandledPromiseRejectionWarning: TypeError: Value is not a valid number: undefined                                  
    at Object.set (/home/node/app/node_modules/prom-client/lib/gauge.js:163:10)                                               
    at /home/node/app/app.js:124:45                                                                                           
    at Array.forEach (<anonymous>)                                                                                            
    at refreshMetrics (/home/node/app/app.js:122:19)                                                                          
    at runMicrotasks (<anonymous>)                                                                                            
    at processTicksAndRejections (internal/process/task_queues.js:97:5)                                                       
(node:1) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an 
async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node pro
cess on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli
_unhandled_rejections_mode). (rejection id: 1133)          

```

Chould you please have a look?

Thanks!
Erik

**Comment by tg44** [comment]
Hy!

This is strange indeed, I saw them in the log, but I watched my grafs and everything seemed ok. I think this is an error on the nicehash part. I had a data change in the last 5 minutes, so it should be ok on the long run.

**Comment by tg44** [comment]
I tried it locally, I can't reproduce this... I think this is some API error on the other side, but ideally, we don't notice this.

**Comment by esille** [comment]
Strange, still broken on my side. Tried to restart container with new combination of API secrets. Same error and same zeros for nh_total_btc and nh_btc_usd_rate. 

Any other ideas what to test/change/share?

**Comment by esille** [comment]
Looking more into code (finally ;) ) and error details: "at refreshMetrics (/home/node/app/app.js:122:19) " 
points to this code `    rigJoinTime.labels(rig.name, rig.rigId).set(rig.joinTime)
`
When I check status of my rig via API call directly there is no joinTime for my rig. Most probably because my RIG is unmanaged (hiveOS). 

You might be able to test for yourself with unmanaged rig. Good idea would be to first check if rig is managed/unmanaged and based on that try to get only relevant data.

Example of my API response for unmanaged rig (GET https://api2.nicehash.com/main/api/v2/mining/rig2/<rigid>)

`{
  "rigId": "Garage",
  "type": "UNMANAGED",
  "name": "Garage",
  "statusTime": 1620994691000,
  "minerStatus": "MINING",
  "unpaidAmount": "0.00000650",
  "stats": [
    {
      "statsTime": 1620994691000,
      "market": "EU",
      "algorithm": {
        "enumName": "DAGGERHASHIMOTO",
        "description": "DaggerHashimoto"
      },
      "unpaidAmount": "0.00000650",
      "difficulty": 0.3880389689172924,
      "proxyId": 1,
      "timeConnected": 1620975702737,
      "xnsub": true,
      "speedAccepted": 183.3984890462453,
      "speedRejectedR1Target": 0,
      "speedRejectedR2Stale": 0,
      "speedRejectedR3Duplicate": 0,
      "speedRejectedR4NTime": 0,
      "speedRejectedR5Other": 0,
      "speedRejectedTotal": 0,
      "profitability": 0.0005328
    }
  ],
  "profitability": 0.0005328,
  "localProfitability": 0
}
`

**Comment by tg44** [comment]
I pushed a fix, I think this should solve your problem, also it adds more "error handling" :D Can you try it after it is out on docker?

**Comment by esille** [comment]
excellent. Now it's showing correct values. Thank you very much for for quick update.

**Comment by tg44** [comment]
I think my newest commit will fix this too. Unmanaged rigs has no device list...

**Comment by esille** [comment]
fixed in #2
